flocker/node/agents/_logging.py

```diff
@@ -94,3 +99,70 @@
     [VOLUME_ID, STATUS, TARGET_STATUS, WAIT_TIME],
     u"Waiting for a volume to reach target status.",)
 # End: Helper datastructures used by AWS storage driver.
+
+# Begin: Helper datastructures used by OpenStack storage drivers
+
+CODE = Field.for_types("code", [int], u"The HTTP response code.")
+MESSAGE = Field.for_types(
+    "message", [bytes, unicode],
+    u"A human-readable error message given by the response.",
+)
+DETAILS = Field.for_types("details", [dict], u"Extra details about the error.")
+REQUEST_ID = Field.for_types(
+    "request_id", [bytes, unicode],
+    u"The unique identifier assigned by the server for this request.",
+)
+URL = Field.for_types("url", [bytes, unicode], u"The request URL.")
+METHOD = Field.for_types("method", [bytes, unicode], u"The request method.")
+
+NOVA_CLIENT_EXCEPTION = MessageType(
+    u"openstack:nova_client_exception", [
+        CODE,
+        MESSAGE,
+        DETAILS,
+        REQUEST_ID,
+        URL,
+        METHOD,
+    ],
+)
+
+RESPONSE = Field.for_types("response", [bytes, unicode], u"The response body.")
+
+KEYSTONE_HTTP_ERROR = MessageType(
+    u"openstack:keystone_http_error", [
+        CODE,
+        RESPONSE,
+        MESSAGE,
+        DETAILS,
+        REQUEST_ID,
+        URL,
+        METHOD,
+    ],
+)
+
+LOCAL_IPS = Field(
+    u"local_ips",
+    repr,
+    u"The IP addresses found on the target node."
+)
+
+API_IPS = Field(
+    u"api_ips",
+    repr,
+    u"The IP addresses and instance_ids for all nodes."
+)
+
+COMPUTE_INSTANCE_ID_NOT_FOUND = MessageType(
+    u"blockdevice:cinder:compute_instance_id:not_found",
+    [LOCAL_IPS, API_IPS],
+    u"Unable to determine the instance ID of this node.",
+)
+
+# ActionType used by OpenStack storage driver.
+OPENSTACK_ACTION = ActionType(
+    u"flocker:node:agents:blockdevice:openstack",
```

Comment:
I'm a bit mixed up by the different message and action types defined here...  `openstack:keystone_http_error`, `blockdevice:cinder:compute_instance_id:not_found`, `flocker:node:agents:blockdevice:openstack`... It looks like three competing naming conventions.  Is this intentional?  If so it probably warrants some explanation in the code.

I know that constructing the names for this field of Eliot isn't really something we've established a good convention for.  I wish we had (and as we add more logging I know that we're going to regret not having one ever more acutely).  I think a lot of the code does try to be at least locally consistent, though - so using a single convention throughout this module would probably be reasonable.  Please defend the values you chose if you chose them for a particular reason, though. :smile: